templates/.gitlab-ci-template.yml

```
@@ -89,20 +89,44 @@ variables:
         if command -v docker-compose &> /dev/null; then
           docker-compose version
         fi
-        
+
         section_end "docker"
       fi
 
       if command -v gcloud &> /dev/null; then
         section_start "gcloud" "Gcloud authentication setup"
+        GCP_SERVICE_ACCOUNT=$(gcloud auth list --filter=status:ACTIVE --format="value(account)")
+
+        # if service account is empty, wait for it to be set.
+        if [ -z "${GCP_SERVICE_ACCOUNT}" ]; then
+          MAX_ATTEMPTS=3
+          i=0
+          while [ $i -lt $MAX_ATTEMPTS ]; do
+            echo "Waiting for gcloud to authenticate..."
+            sleep 5s
+            GCP_SERVICE_ACCOUNT=$(gcloud auth list --filter=status:ACTIVE --format="value(account)")
+            if [ -n "${GCP_SERVICE_ACCOUNT}" ]; then
+              break
+            fi
+            i=$((i+1))
+          done
+        fi
+
+        # if still empty, fail.
+        if [ -z "${GCP_SERVICE_ACCOUNT}" ]; then
+          echo "Failed to authenticate with gcloud after multiple attempts."
+          exit 1
+        fi
```

```suggestion
      if [ "${DISABLE_GCP_SERVICE_ACCOUNT_CHECK:-0}" != "1" ]; then
        GCP_SERVICE_ACCOUNT=$(gcloud auth list --filter=status:ACTIVE --format="value(account)")

        # if service account is empty, wait for it to be set.
        if [ -z "${GCP_SERVICE_ACCOUNT}" ]; then
          MAX_ATTEMPTS=3
          i=0
          while [ $i -lt $MAX_ATTEMPTS ]; do
            echo "Waiting for gcloud to authenticate..."
            sleep 5s
            GCP_SERVICE_ACCOUNT=$(gcloud auth list --filter=status:ACTIVE --format="value(account)")
            if [ -n "${GCP_SERVICE_ACCOUNT}" ]; then
              break
            fi
            i=$((i+1))
          done
        fi

        # if still empty, fail.
        if [ -z "${GCP_SERVICE_ACCOUNT}" ]; then
          echo "Failed to authenticate with gcloud after multiple attempts."
          exit 1
        fi
      fi
```

I would prefer to have a method to skip this check. What do you think?